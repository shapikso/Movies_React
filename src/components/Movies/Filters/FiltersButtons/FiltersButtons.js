import React from 'react';
import Button from "../../../common/Button/Button";
import PropTypes from "prop-types";

const FiltersButtons = ({formSubmitHandler, formResetHandler, closeModal}) =>
    (
        <div className="form-filters__buttons">
            <Button onClick={formSubmitHandler} className="basic-btn" contentKey ="Submit"/>
            <Button onClick={formResetHandler} type="reset" className="basic-btn"  contentKey="Reset"/>
            <span onClick={closeModal} className="close">x</span>
        </div>
    );

FiltersButtons.propTypes = {
    formSubmitHandler: PropTypes.func.isRequired,
    formResetHandler : PropTypes.func.isRequired,
    closeModal: PropTypes.func.isRequired,
};

export default FiltersButtons;