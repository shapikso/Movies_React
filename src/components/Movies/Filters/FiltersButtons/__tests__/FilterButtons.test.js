import React from 'react';
import {shallow, mount} from 'enzyme';
import FiltersButtons from "../FiltersButtons";

describe('FiltersButtons', ()=> {
    const props = {
        formSubmitHandler: jest.fn(),
        formResetHandler: jest.fn(),
        isResetDisabled: true,
    };
    it('should render correctly', () => {
        const component = shallow(<FiltersButtons {...props} />);
        expect(component).toMatchSnapshot();
    });
    it('should checked render isResetDisabled', () => {
        const component = mount(<FiltersButtons {...props} />);
        expect(component.find('button').at(0).getElement().props.disabled).toEqual(props.isResetDisabled);
    });
    it('should reset form on click', () => {
        const component = mount(<FiltersButtons {...props} />);
        component.find('button').at(0).getElement().props.onClick();
        expect(props.formResetHandler).toHaveBeenCalled();
    });
    it('should submit form on click', () => {
        const component = mount(<FiltersButtons {...props} />);
        component.find('button').at(1).getElement().props.onClick();
        expect(props.formSubmitHandler).toHaveBeenCalled();
    });
});