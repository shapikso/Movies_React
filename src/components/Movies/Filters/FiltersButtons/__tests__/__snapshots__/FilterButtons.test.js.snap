// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersButtons should render correctly 1`] = `ShallowWrapper {}`;
